vit-servicing-station-tests/src/common/data/generator/voting/builder/mod.rs

```diff
@@ -0,0 +1,135 @@
+mod challenge;
+mod proposal;
+
+use crate::common::data::ArbitraryValidVotingTemplateGenerator;
+use crate::common::data::ChallengeTemplate;
+use crate::common::data::FundTemplate;
+use crate::common::data::ProposalTemplate;
+use crate::common::data::ReviewTemplate;
+use crate::common::data::ValidVotingTemplateGenerator;
+pub use challenge::ChallengeBuilder;
+pub use proposal::ProposalBuilder;
+
+#[derive(Clone, Default)]
+pub struct ArbitraryValidVotePlanBuilder {
+    template_generator: ArbitraryValidVotingTemplateGenerator,
+    challenges: Vec<ChallengeBuilder>,
+}
+
+impl ArbitraryValidVotePlanBuilder {
+    pub fn challenges(mut self, challenges: Vec<ChallengeBuilder>) -> Self {
+        for challenge in challenges.into_iter() {
+            self = self.challenge(challenge);
+        }
+        self
+    }
+
+    pub fn challenge(mut self, mut challenge: ChallengeBuilder) -> Self {
+        challenge
```

Comment:
I am wondering why it is needed modify `challenge` variable before pushing it into `self. challenges`, don't we expect that it should be correctly initialized outside this function, somewhere inside `ProposalBuilder` methods

Reply:
Very good point. My priority was to simplify config construction to maximum. Joining ids is tiresome process so wanted to move it to internals. My goal was that person who creates config only defines business parameters:

```
       let mut template = ArbitraryValidVotePlanConfig::default().challenge(
            ChallengeConfig::default()
                .rewards_total(1000)
                .proposers_rewards(1000)
                .proposal(ProposalConfig::default().funds(100)),
        );
```

Reply:
Yes, I understood, but I think it will be better to move it inside `ProposalConfig::default()` function, or it is not in place of definition of the `default` function for the `ProposalConfig` ?

Reply:
I managed to move assigning challenges id using lazy static and proposals when adding them to challenge

Reply:
Hmm, I think in this case previous implementation will be better, so lets revert it